src/main.cc

```diff
@@ -288,11 +288,211 @@ NAN_METHOD(EnumKeys) {
     RegCloseKey(hCurrentKey);
 }
 
-NAN_MODULE_INIT(Init) {
+NAN_METHOD(CreateKey)
+{
+  auto argCount = info.Length();
+  if (argCount != 2)
+  {
+    Nan::ThrowTypeError("Wrong number of arguments");
+    return;
+  }
+
+  if (!info[0]->IsNumber())
+  {
+    Nan::ThrowTypeError("A number was expected for the first argument, but wasn't received.");
+    return;
+  }
+
+  if (!info[1]->IsString())
+  {
+    Nan::ThrowTypeError("A string was expected for the second argument, but wasn't received.");
+    return;
+  }
+
+  auto first = reinterpret_cast<HKEY>(Nan::To<int64_t>(info[0]).FromJust());
+
+  HKEY hCurrentKey = first;
+  if (!info[1]->IsNullOrUndefined())
+  {
+    Nan::Utf8String subkeyArg(Nan::To<v8::String>(info[1]).ToLocalChecked());
+    auto subKey = utf8ToWideChar(std::string(*subkeyArg));
+    if (subKey == nullptr)
+    {
+      Nan::ThrowTypeError("A string was expected for the second argument, but could not be parsed.");
+      return;
+    }
+    auto newKey = RegCreateKeyEx(
+        first,
+        subKey,
+        0,
+        nullptr,
+        REG_OPTION_NON_VOLATILE,
+        KEY_READ | KEY_WOW64_64KEY,
+        nullptr,
+        &hCurrentKey,
+        nullptr);
+    if (newKey != ERROR_SUCCESS)
+    {
+      // FIXME: the key does not exist, just return an empty array for now
+      info.GetReturnValue().Set(New<v8::Boolean>(false));
+      return;
+    }
+  }
+
+  info.GetReturnValue().Set(New<v8::Boolean>(true));
+  if (hCurrentKey != first)
+    RegCloseKey(hCurrentKey);
+}
+
+NAN_METHOD(SetValue)
+{
+  {
+    auto argCount = info.Length();
+    if (argCount != 5)
+    {
+      Nan::ThrowTypeError("Wrong number of arguments");
+      return;
+    }
+
+    if (!info[0]->IsNumber())
+    {
+      Nan::ThrowTypeError("A number was expected for the first argument, but wasn't received.");
+      return;
+    }
+
+    if (!info[1]->IsString())
+    {
+      Nan::ThrowTypeError("A string was expected for the second argument, but wasn't received.");
+      return;
+    }
+
+    if (!info[2]->IsString())
+    {
+      Nan::ThrowTypeError("A string was expected for the third argument, but wasn't received.");
+      return;
+    }
+
+    if (!info[3]->IsString())
+    {
+      Nan::ThrowTypeError("A string was expected for the fourth argument, but wasn't received.");
+      return;
+    }
+
+    if (!info[4]->IsString())
+    {
+      Nan::ThrowTypeError("A string was expected for the fourth argument, but wasn't received.");
```

Comment:
I just noticed this minor typo:
```suggestion
      Nan::ThrowTypeError("A string was expected for the fifth argument, but wasn't received.");
```